pkg/trainer/tensorboard_test.go

```diff
@@ -0,0 +1,113 @@
+package trainer
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+
+	"reflect"
+	"sync"
+
+	"github.com/jlewi/mlkube.io/pkg/spec"
+	tfJobFake "github.com/jlewi/mlkube.io/pkg/util/k8sutil/fake"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+func TestTBReplicaSet(t *testing.T) {
+	clientSet := fake.NewSimpleClientset()
+
+	jobSpec := &spec.TfJob{
+		Spec: spec.TfJobSpec{
+			RuntimeId: "some-runtime",
+			ReplicaSpecs: []*spec.TfReplicaSpec{
+				{
+					Replicas:      proto.Int32(1),
+					TfPort:        proto.Int32(10),
+					Template:      &v1.PodTemplateSpec{},
+					TfReplicaType: spec.MASTER,
+				},
+			},
+			TensorBoard: &spec.TensorBoardSpec{
+				LogDir: "/tmp/tensorflow",
+			},
+		},
+	}
+
+	stopC := make(chan struct{})
+
+	wg := &sync.WaitGroup{}
+	job, err := initJob(clientSet, &tfJobFake.TfJobClientFake{}, jobSpec, stopC, wg)
+
+	if err != nil {
+		t.Fatalf("initJob failed: %v", err)
+	}
+
+	replica, err := NewTBReplicaSet(clientSet, *jobSpec.Spec.TensorBoard, job)
+
+	if err != nil {
+		t.Fatalf("NewTBReplicaSet failed: %v", err)
+	}
+
+	if err := replica.Create(); err != nil {
+		t.Fatalf("TBReplicaSet.Create() error; %v", err)
+	}
+
+	// Expected labels
+	expectedLabels := map[string]string{
+		"mlkube.io":  "",
+		"app":        "tensorboard",
+		"runtime_id": "some-runtime",
+	}
+
+	// Check that a service was created.
+	sList, err := clientSet.CoreV1().Services(NAMESPACE).List(meta_v1.ListOptions{})
```

Comment:
No, my bad I should change that.
I'm currently in the plane so I added a Todo as I won't have time to test.